lib/data_freshness.rb

```diff
@@ -0,0 +1,53 @@
+require 'datadog/statsd'
+require 'aws-sdk-s3'
+require_relative './config'
+
+# Records timeliness of data-processing tasks' results on S3
+class DataFreshness
+  S3_LOCATIONS = {
+    artist_career_stage: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtistCareerStage/' },
+    artist_gene_value: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtistGeneValue/' },
+    artist_related_genes: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtistRelatedGenes/' },
+    artist_similarity: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtistSimilarity/' },
+    artist_trending: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtistTrending/' },
+    artwork_gene_value: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtworkGeneValue/' },
+    artwork_iconicity: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtworkIconicity/' },
+    artwork_merchandisability: { bucket: 'artsy-cinder-production', prefix: 'builds/ArtworkMerchandisability/' },
+    cf_artwork_similarity: { bucket: 'artsy-cinder-production', prefix: 'builds/CFArtworkSimilarity/' },
+    contemporary_artist_similarity: { bucket: 'artsy-cinder-production', prefix: 'builds/ContemporaryArtistSimilarity/' },
+    gene_partitioned_artist_trending: { bucket: 'artsy-cinder-production', prefix: 'builds/GenePartitionedArtistTrending/' },
+    gene_similarity: { bucket: 'artsy-cinder-production', prefix: 'builds/GeneSimilarity/' },
+    sitemaps: { bucket: 'artsy-sitemaps', prefix: 'sitemap-artist-series' }, # This is because in hue artist series is the last task to run in the chained sitemap tasks
```

Comment:
This isn't necessarily static and could change without folks realizing there's a relationship between the order and these metrics. Would be worth thinking about what we could do to make the relationship more explicit.

Reply:
Yup totally. Our initial implementation simply matched any new sitemap uploads, but the latest failure comes part-way through the job, so auction sitemaps (e.g.) were updated while those later in the workflow (like artist series) were stale.

We could even declare separate prefixes and measure freshness of the different groups of sitemaps with separate metrics.